### PR TITLE
adding logic to handle existing files and improving logging

### PR DESCRIPTION
**Pull request by nthain**

### CODE REVIEW

File: ingest_revisions_dataflow/dataflow_main.py [hunk 1]
```diff
@@ -24,25 +33,16 @@ def run(argv=None):
                       dest='output',
                       # CHANGE 1/5: The Google Cloud Storage path is required
                       # for outputting the results.
-                      default='ingested/',
+                      default='gs://wikidetox-viz-dataflow/output_lists/ingestion_output.txt',
```
Comment (iislucas) [inline ingest_revisions_dataflow/dataflow_main.py]: I'd rather leave project names out of code, and have a JSON config file that is not checked in. This allows other to more naturally and easily re-run our work.

File: ingest_revisions_dataflow/dataflow_main.py [hunk 2]
```diff
@@ -53,40 +53,61 @@ def run(argv=None):
...
 class WriteDecompressedFile(beam.DoFn):
   def process(self, element):
     logging.info('USERLOG: Working on %s' % element)
     chunk_name = element
-
-    CLOUD_STORAGE_BUCKET = 'wikidetox-viz-dataflow' # Should be deletable
-    OUT_PATH = 'ingested/' # Should be deletable
+    status = 'NO STATUS'
 
     in_file_path = 'gs://wikidetox-viz-dataflow/raw-downloads/' + chunk_name
```
Comment (iislucas) [inline ingest_revisions_dataflow/dataflow_main.py]: Better to use path.join.

File: ingest_revisions_dataflow/dataflow_main.py [hunk 3]
```diff
@@ -53,40 +53,61 @@ def run(argv=None):
...
+        cp_local_cmd = (['gsutil', 'cp', in_file_path, './'])
+        subprocess.call(cp_local_cmd)
+
+        logging.info('USERLOG: Loading ingester with input: %s output: %s' % (chunk_name, local_out_filename))
+        ingester = wiki_ingester.WikipediaRevisionsIngester(chunk_name, local_out_filename)
+        logging.info('USERLOG: Running ingester on %s.' % chunk_name)
+        ingester.run_ingester()
+      
+        logging.info('USERLOG: Running gsutil cp %s %s' % ('./' + local_out_filename, out_file_path))
+        cp_remote_cmd = (['gsutil', 'cp', './' + local_out_filename, out_file_path])
```
Comment (iislucas) [inline ingest_revisions_dataflow/dataflow_main.py]: Again, better to use path library than string concat.

**Comment by nthain** [comment]
Thanks for the suggestions. I'll make the changes to the code.

**Comment by nthain** [comment]
Suggestions should be fixed in the most recent commit.

File: ingest_revisions_dataflow/dataflow_main.py [hunk 4]
```diff
@@ -62,11 +58,11 @@ def process(self, element):
     chunk_name = element
     status = 'NO STATUS'
 
-    in_file_path = 'gs://wikidetox-viz-dataflow/raw-downloads/' + chunk_name
+    in_file_path = path.join('gs://wikidetox-viz-dataflow/raw-downloads/', chunk_name)
```
Comment (iislucas) [inline ingest_revisions_dataflow/dataflow_main.py]: This project path reference should be moved into the config too. :)

**Pull request merged**